docs/en/concepts-and-designs/lal.md

```diff
@@ -0,0 +1,239 @@
+
+#### `yaml`
+
+<!-- TODO: is structured in the reported (gRPC) `LogData`, not much to do -->
+
+#### `text`
+
+For unstructured logs, there are some `text` parsers for use.
+
+- `regexp`
+
+`regexp` parser uses a regular expression (`regexp`) to parse the logs. It leverages the captured groups of the regexp,
+all the captured groups can be used later in the extractors or sinks.
+
+```groovy
+filter {
+    text {
+        regexp "(?<timestamp>\\d{8}) (?<thread>\\w+) (?<level>\\w+) (?<traceId>\\w+) (?<msg>.+)"
+        // this is just a demo pattern
+    }
+    extractor {
+        tag key: "level", val: parsed.level
+        // we add a tag called `level` and its value is parsed.level, captured from the regexp above
+        traceId parsed.traceId
+        // we also extract the trace id from the parsed result, which will be used to associate the log with the trace
+    }
+    // ...
+}
+```
+
+- `grok`
+
+<!-- TODO: grok Java library has poor performance, need to benchmark it, the idea is basically the same with `regexp` above -->
+
+### Extractor
+
+Extractors aim to extract metadata from the logs. The metadata can be a service name, a service instance name, an
+endpoint name, or even a trace ID, all of which can be associated with the existing traces and metrics.
+
+- `service`
+
+`service` extracts the service name from the `parsed` result, and set it into the `LogData`, which will be persisted (if
+not dropped) and is used to associate with traces / metrics.
+
+- `instance`
+
+`instance` extracts the service instance name from the `parsed` result, and set it into the `LogData`, which will be
+persisted (if not dropped) and is used to associate with traces / metrics.
+
+- `endpoint`
+
+`endpoint` extracts the service instance name from the `parsed` result, and set it into the `LogData`, which will be
+persisted (if not dropped) and is used to associate with traces / metrics.
+
+- `traceId`
+
+`traceId` extracts the trace ID from the `parsed` result, and set it into the `LogData`, which will be
+persisted (if not dropped) and is used to associate with traces / metrics.
+
+- `segmentId`
+
+`segmentId` extracts the segment ID from the `parsed` result, and set it into the `LogData`, which will be
+persisted (if not dropped) and is used to associate with traces / metrics.
+
+- `spanId`
+
+`spanId` extracts the span ID from the `parsed` result, and set it into the `LogData`, which will be
+persisted (if not dropped) and is used to associate with traces / metrics.
+
+- `metrics`
+
+`metrics` extracts / generates metrics from the logs, and sends the generated metrics to the meter system, you can configure [MAL](mal.md) for further analysis of these metrics. Examples are as follows:
+
+```groovy
+filter {
+    // ...
+    extractor {
+        service parsed.serviceName
+        metrics("log_count") {
+            timestamp: parsed.timestamp
+            tags: ["level": parsed.level, "service": parsed.service, "instance": parsed.instance]
```

Comment:
Could we use `parsed.*` as tag's `key`?

Reply:
> Could we use `parsed.*` as tag's `key`?

Yes, you can use `tags: [(parsed.foo): parsed.bar]` or `tags: ["$parsed.foo": parsed.bar]`

Reply:
We should update this on the document.